### PR TITLE
Created HostIpXml all alternative to HostIp

### LINKED ISSUE #1056: HostIP no longer supplies lat,lng in JSON
It's probably due that MaxMind commercialized it or  due to GDPR
http://api.hostip.info/get_json.php
but XML still has it http://api.hostip.info/get_xml.php

### PR DESCRIPTION
**Pull request by oleg-andreyev**
fixes #1056

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -41,7 +41,9 @@
         "ext-geoip": "Enabling the geoip extension allows you to use the MaxMindProvider.",
         "geoip/geoip": "If you are going to use the MaxMindBinaryProvider (conflict with geoip extension).",
         "geoip2/geoip2": "If you are going to use the GeoIP2DatabaseProvider.",
-        "symfony/stopwatch": "If you want to use the TimedGeocoder"
+        "symfony/stopwatch": "If you want to use the TimedGeocoder",
+        "ext-simplexml": "If you want to use HostIpXml",
+        "ext-json": "If you want to use HostIp"
```
Comment (jbelien) [inline composer.json:46]: Please don't touch the main `composer.json` file.
Use `composer.json` file from provider.

File: src/Provider/HostIp/AbstractHostIp.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+use Geocoder\Model\Address;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+
+/**
+ * @author William Durand <william.durand1@gmail.com>
+ * @author Oleg Andreyev <oleg@andreyev.lv>
+ */
+abstract class AbstractHostIp extends AbstractHttpProvider
```
Comment (jbelien) [inline src/Provider/HostIp/AbstractHostIp.php:27]: The class needs to implement `Geocoder\Provider\Provider`

File: src/Provider/HostIp/HostIp.php [hunk 3]
```diff
@@ -76,42 +38,15 @@ public function getName(): string
      *
      * @return Collection
      */
-    private function executeQuery(string $url): AddressCollection
+    protected function executeQuery(string $url): AddressCollection
     {
         $content = $this->getUrlContents($url);
-        $data = json_decode($content, true);
+        $data = \json_decode($content, true);
```
Comment (jbelien) [inline src/Provider/HostIp/HostIp.php:44]: The `\` is not necessary.
```suggestion
        $data = json_decode($content, true);
```

File: src/Provider/HostIp/HostIpXml.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    /**
+     * @param string $url
+     *
+     * @return Collection
+     */
+    protected function executeQuery(string $url): AddressCollection
+    {
+        $content = $this->getUrlContents($url);
+        $xml = \simplexml_load_string($content);
```
Comment (jbelien) [inline src/Provider/HostIp/HostIpXml.php:44]: The `\` is not necessary.

```suggestion
        $xml = simplexml_load_string($content);
```

**Review by jbelien — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (oleg-andreyev) [reply composer.json:46]: it's a "suggest" section - done it similar to `geoip2/geoip2`
composer.json from a provider will have a require

File: src/Provider/HostIp/AbstractHostIp.php (hunk 2 above)
Comment (oleg-andreyev) [reply src/Provider/HostIp/AbstractHostIp.php:27]: it's already implemented by \Geocoder\Provider\AbstractProvider

`AbstractHttpProvider -> AbstractProvider -> Provider`

File: src/Provider/HostIp/HostIp.php (hunk 3 above)
Comment (oleg-andreyev) [reply src/Provider/HostIp/HostIp.php:44]: replaced with `use`

File: src/Provider/HostIp/HostIpXml.php (hunk 4 above)
Comment (oleg-andreyev) [reply src/Provider/HostIp/HostIpXml.php:44]: replaced with `use`

File: src/Provider/HostIp/AbstractHostIp.php (hunk 2 above)
Comment (jbelien) [reply src/Provider/HostIp/AbstractHostIp.php:27]: I do know that ; this is just for consistency !

Comment (jbelien) [reply src/Provider/HostIp/AbstractHostIp.php:27]: Indeed ; though, could you add it for consistency with the other providers ?

Comment (oleg-andreyev) [reply src/Provider/HostIp/AbstractHostIp.php:27]: ok.

**Comment by jbelien** [comment]
Thanks @oleg-andreyev !
New release will be published during the day 🎉

**Pull request merged**